# Description: add launchpad menu items to the application
# UbuntuSpecific: launchpad
#
diff -Nur -x '*.orig' -x '*~' eog-2.22.1/configure.ac eog-2.22.1.new/configure.ac
--- eog-2.22.1/configure.ac	2008-04-07 21:04:28.000000000 +0200
+++ eog-2.22.1.new/configure.ac	2008-04-09 11:11:24.000000000 +0200
@@ -99,6 +99,7 @@
              libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED \
              libglade-2.0 >= $LIBGLADE_REQUIRED \
              gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED \
+	     launchpad-integration \
              gconf-2.0 >= $GCONF_REQUIRED \
              libart-2.0 >= $LIBART_REQUIRED \
              gnome-desktop-2.0 >= $GNOME_DESKTOP_REQUIRED \
diff -Nur -x '*.orig' -x '*~' eog-2.22.1/data/eog-ui.xml eog-2.22.1.new/data/eog-ui.xml
--- eog-2.22.1/data/eog-ui.xml	2008-04-07 21:04:12.000000000 +0200
+++ eog-2.22.1.new/data/eog-ui.xml	2008-04-09 11:11:34.000000000 +0200
@@ -69,6 +69,7 @@
     </menu>
     <menu action="Help">
       <menuitem action="HelpManual"/>
+      <placeholder name="LaunchpadItems"/>
       <menuitem action="HelpAbout"/>
     </menu>
   </menubar>
diff -Nur -x '*.orig' -x '*~' eog-2.22.1/src/eog-window.c eog-2.22.1.new/src/eog-window.c
--- eog-2.22.1/src/eog-window.c	2008-04-07 21:03:38.000000000 +0200
+++ eog-2.22.1.new/src/eog-window.c	2008-04-09 11:11:24.000000000 +0200
@@ -67,6 +67,7 @@
 #include <libgnomevfs/gnome-vfs-mime.h>
 #include <libgnomevfs/gnome-vfs-mime-handlers.h>
 #include <gconf/gconf-client.h>
+#include <launchpad-integration.h>
 
 #if HAVE_LCMS
 #include <X11/Xlib.h>
@@ -3904,6 +3905,9 @@
                 g_error_free (error);
         }
 
+	launchpad_integration_add_ui (priv->ui_mgr,
+				      "/MainMenu/Help/LaunchpadItems");
+
 	g_signal_connect (priv->ui_mgr, "connect_proxy",
 			  G_CALLBACK (connect_proxy_cb), window);
 	g_signal_connect (priv->ui_mgr, "disconnect_proxy",
